File: WEEK2/bj6_10989.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0]);
const arr = [];
for (let i = 1; i <= n; i++) {
    arr.push(parseInt(input[i]));
}

function radixSort(arr) {
    const maxVal = Math.max(...arr);
    const maxDigit = String(maxVal).length;

    let radix = 1;
    for (let i = 0; i < maxDigit; i++) {
        const buckets = Array.from({ length: 10 }, () => []);
        for (let j = 0; j < arr.length; j++) {
            const digit = Math.floor(arr[j] / radix) % 10;
            buckets[digit].push(arr[j]);
        }

        arr = [].concat(...buckets);
        radix *= 10;
    }
    return arr;
}

const sortedArr = radixSort(arr);
console.log(sortedArr.join('\n'));

File: WEEK2/bj6_10989.py
import sys


def radix_sort(arr):
    max_val = max(arr)
    max_digit = len(str(max_val))

    radix = 1

    for _ in range(max_digit):
        buckets = [[] for _ in range(10)]
        for num in arr:
            digit = (num // radix) % 10
            buckets[digit].append(num)

        arr = []
        for bucket in buckets:
            arr.extend(bucket)

        radix *= 10
    return arr


n = int(sys.stdin.readline())
arr = []
for _ in range(n):
    arr.append(int(sys.stdin.readline()))

sorted_arr = radix_sort(arr)

for val in sorted_arr:
    print(val)
